app/routes.py

```diff
@@ -1,139 +1,165 @@
     post = Post(body=body, user_id=user_id)
     try:
         post.save()
-        return jsonify({"title":"Post saved", "msg":"Post saved to database"}),200
+        return jsonify({"title": "Post saved", "msg": "Post saved to database"}), 200
     except Exception as error:
         db.session.rollback()
-        return jsonify({"title":"save failed", "msg": str(error)}), 400
+        return jsonify({"title": "save failed", "msg": str(error)}), 400
 
 
 # Get all posts from db
-@app.route('/posts', methods=['GET'])
+@app.route("/posts", methods=["GET"])
 def manage_posts():
-    posts = Post.query.all()
-    return jsonify([{"body":post.body, "user_id":post.user_id, "id": post.id} for post in posts]), 200
+    posts = Post.get_all()
+    return (
+        jsonify(
+            [
+                {"body": post.body, "user_id": post.user_id, "id": post.id}
+                for post in posts
+            ]
+        ),
+        200,
+    )
 
 
 # Create, Update, Get, Delete one user from db
-@app.route('/user', methods=['POST'])
+@app.route("/user", methods=["POST"])
 def add_user():
     try:
         data = request.get_json(force=True)
     except Exception as error:
-        return jsonify({"title":"request failed", "msg": str(error)}), 400
+        return jsonify({"title": "request failed", "msg": str(error)}), 400
     username = data.get("username")
     password = data.get("password")
     if not username or not password:
-        return jsonify({"title":"Save failed","msg":"username and password are required"})
+        return (
+            jsonify(
+                {"title": "Save failed", "msg": "username and password are required"}
+            ),
+            400,
+        )
     user = User(username=username)
     user.set_password(password)
     try:
         user.save()
         return jsonify({"title": "User saved", "msg": "User saved to db"}), 201
     except Exception as error:
         db.session.rollback()
-        return jsonify({"title":"save failed", "msg": str(error)}), 400
+        return jsonify({"title": "save failed", "msg": str(error)}), 400
 
 
-@app.route('/user/<user_id>', methods=['GET', 'PUT', 'DELETE'])
+@app.route("/user/<user_id>", methods=["GET", "PUT", "DELETE"])
 def manage_user(user_id):
     if request.method == "PUT":
         try:
             data = request.get_json(force=True)
         except Exception as error:
-            return jsonify({"title":"request failed", "msg": str(error)}), 400
+            return jsonify({"title": "request failed", "msg": str(error)}), 400
         username = data.get("username")
         password = data.get("password")
         if not username or not password:
-            return jsonify({"title":"Save failed","msg":"username and password are required"})
-        user = User.query.get(user_id)
+            return jsonify(
+                {"title": "Save failed", "msg": "username and password are required"}
+            )
+        user = User.get_by_id(user_id)
```

Comment:
If user doesn't exist you will get an error because you will try to do `Nonetype.username = username`